lib/engine/game/g_18_carolinas.rb

```diff
@@ -7,6 +7,37 @@ module Engine
   module Game
     class G18Carolinas < Base
       load_from_json(Config::Game::G18Carolinas::JSON)
+
+      def init_companies(players)
+        super.take(players.size)
+      end
+
+      def setup
+        @companies.each_with_index do |company, index|
+          price = company.value
+          player = @players[index]
+          company.owner = player
+          @players[index].companies << company
+          @players[index].spend(price, @bank)
+          @log << "#{player.name} buys #{company.name} for #{format_currency(price)}"
+        end
+
+        @corporations[0..3].sort_by! { @game.rand }
+        @corporations[4..7].sort_by! { @game.rand }
+
+        @corporations.each_with_index do |corporation, index|
+          index -= 4 if index > 3
```

Comment:
what are you trying to do here? modulo? if so then it should be like index % 4

Reply:
This code sets the par prices randomly to start in each of two groups (NC corps and SC corps).
First we randomized positions 0 to 3 (NC), then 4 to 7 (SC).
The index is used to set the par prices for each group 90/80/70/60. Do you think I should do it in a different way?